npm/pod.go

```diff
@@ -0,0 +1,140 @@
+package npm
+
+import (
+	"strings"
+
+	"github.com/Azure/azure-container-networking/log"
+	"github.com/Azure/azure-container-networking/npm/util"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+func isValidPod(podObj *corev1.Pod) bool {
+	return podObj.Status.Phase != "Failed" &&
+		podObj.Status.Phase != "Succeeded" &&
+		podObj.Status.Phase != "Unknown" &&
+		len(podObj.Status.PodIP) > 0
+}
+
+func isSystemPod(podObj *corev1.Pod) bool {
+	return podObj.ObjectMeta.Namespace == util.KubeSystemFlag
+}
+
+// AddPod handles add pod.
+func (npMgr *NetworkPolicyManager) AddPod(podObj *corev1.Pod) error {
```

Comment:
Should the name of the method be more descriptive? like AddPodToIpSet or something?

Reply:
addressed